src/pps/pps.proto

```diff
@@ -989,6 +990,127 @@ message SetProjectDefaultsResponse {
   repeated Pipeline affected_pipelines = 1;
 }
 
+message LogQuery {
+    oneof query_type {
+        UserLogQuery user = 1;
+        AdminLogQuery admin = 2;
+    }
+}
+
+message AdminLogQuery {
+    oneof admin_type {
+        string logql = 1; // Arbitrary LogQL query.
+        string pod = 2; // A pod's logs (all containers).
+        PodContainer pod_container = 3; // One container.
+        string app = 4; // One "app" (logql -> {app=X})
+        PipelineLogQuery master = 5; // All master worker lines from a pipeline.
+        PipelineLogQuery storage = 6; // All storage container lines from a pipeline.
+        UserLogQuery user = 7; // All worker lines from a pipeline/job.
+    }
+}
+
+message PodContainer {
+    string pod = 1;
+    string container = 2;
+}
+
+enum LogLevel {
```

Comment:
In the matchers this gets translated to a uint32; the matcher should take this LogLevel type instead.